### PR TITLE
RHCLOUD-36814: Log when nothing to replicate for disabled user

### PR DESCRIPTION
**Pull request by alechenninger**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-36814

## Description of Intent of Change(s)

Log when nothing to replicate for disabled to user, instead of publishing an empty replication event.

An empty replication event is always considered a bug and should be prevented more explicitly in the surrounding code, so it's clear why it is empty (and therefore clear we're not missing replication unintentionally.)

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [x] if API spec changes are required, is the spec updated?
- [x] are there any pre/post merge actions required? if so, document here.
- [x] are theses changes covered by unit tests?
- [x] if warranted, are documentation changes accounted for?
- [x] does this require migration changes?
  - [x] if yes, are they backwards compatible?
- [x] is there known, direct impact to dependent teams/components?
  - [x] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/management/tenant_service/v2.py [hunk 1]
```diff
@@ -206,34 +206,52 @@ def _disable_user_in_tenant(self, user: User):
...
 
+        logger.info(
+            f"Removing Principal and group membership from RBAC and Relations. user_id={user_id} org_id={user.org_id}"
+        )
+
         try:
             mapping = TenantMapping.objects.filter(tenant__org_id=user.org_id).get()
-            tuples_to_remove.append(Group.relationship_to_user_id_for_group(str(mapping.default_group_uuid), user_id))
+            default_group_uuid = str(mapping.default_group_uuid)  # type: ignore
+            tuples_to_remove.append(Group.relationship_to_user_id_for_group(default_group_uuid, user_id))
```
Comment (alechenninger) [inline rbac/management/tenant_service/v2.py:222]: TODO: shouldn't we also remove the admin group membership tuple?

**Review by astrozzc — APPROVED** [review]
Yea, that admin_group relationship seems to be missing

**Comment by alechenninger** [comment]
I think we may be missing removing the custom group memberships too 🤔

**Comment by astrozzc** [comment]
> I think we may be missing removing the custom group memberships too 🤔

It has been taken care by this, right? tuple = group.relationship_to_principal(user)

**Pull request merged**